collections/soldiers.js

```diff
@@ -0,0 +1,36 @@
+const MongoClient = require('mongodb').MongoClient;
+const assert = require('assert');
+const configs = require('../configFile');
+
+// Connection URL
+const url = configs.dbUrl;
+
+// Database Name
+const dbName = configs.dbName;
+
+// Use connect method to connect to the server
+function newSoldier(db, data, callback) {
+  if (!(/[0-9]{7}/.test(data["id"]))) {
```

Comment:
you use this check multiple times, encapsulate it in a method and add it to soldier utilities